### DISCUSSION TITLE
Displaying per-page banner depending on a value of specific front matter field?

### DISCUSSION THREAD

**Discussion by allenscha** [facebook/docusaurus#9645] [Q&A] [answered]
We're managing multi-language websites, where content in some languages are updated after content in other languages, due to translation and validation schedule.

For pages in such languages, the content is either partially translated or not translated at all, so we're trying to add a banner that says something like "All or part of content in this page is still being translated".

Is there any way to achieve this with front matter field, instead of adding banners manually?

**Comment by slorber** [comment, marked as answer]
You can use custom front matter on your site an swizzle the docs components to read your custom front matter and present a banner.


I'd recommend to swizzle the `DocItem/Layout` for that.

```tsx
export default function DocItemLayout({children}: Props): JSX.Element {
  const docTOC = useDocTOC();
  const {
    metadata: {unlisted, frontMatter},
  } = useDoc();
  return (
    <div className="row">
      <div className={clsx('col', !docTOC.hidden && styles.docItemCol)}>
        {unlisted && <Unlisted />}

        {frontMatter.is_partially_translated && (
          <MyPartiallyTranslatedBanner/>
        )}
        
        <DocVersionBanner />
        <div className={styles.docItemContainer}>
          <article>
            <DocBreadcrumbs />
            <DocVersionBadge />
            {docTOC.mobile}
            <DocItemContent>{children}</DocItemContent>
            <DocItemFooter />
          </article>
          <DocItemPaginator />
        </div>
      </div>
      {docTOC.desktop && <div className="col col--3">{docTOC.desktop}</div>}
    </div>
  );
}
```

But it really depends where you want to display that banner in the layout. 

Wrapping an existing component instead of ejecting `DocItem/Layout` can reduce the amount of code to maintain.
